### PR TITLE
Adding migrated GCP checkov rules to GCP terraform checks - Part 3

### PR DESCRIPTION
**Pull request by mertcancoskuner**
This is part 3 of addressing https://github.com/returntocorp/semgrep-rules/issues/1694

### CODE REVIEW

File: terraform/gcp/security/gcp-org-impersonation-roles-iam-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+rules:
+    - id: gcp-org-impersonation-roles-iam-binding
+      patterns:
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_organization_iam_binding" "..." {
+              ...
+              role = "roles/editor"
+              members = [ ... ]
```
Comment (minusworld) [inline terraform/gcp/security/gcp-org-impersonation-roles-iam-binding.yaml:9]: Based on the message written, it seems like the issue this rule tries to find is where `role = "roles/editor"` and one of the members is a service account. Is this is case, or is the only thing that's important to find the fact that `roles/editor` is used?

File: terraform/gcp/security/gcp-org-impersonation-roles-iam-member.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+rules:
+    - id: gcp-org-impersonation-roles-iam-member
+      patterns:
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_organization_iam_member" "..." {
+              ...
+              role = "roles/editor"
+              member = ...
```
Comment (minusworld) [inline terraform/gcp/security/gcp-org-impersonation-roles-iam-member.yaml:9]: Same question as above

File: terraform/gcp/security/gcp-org-member-default-service-account-iam-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_organization_iam_binding" "..." {
+              ...
+              members = [..., $MEMBER, ...]
+              ...
+              }
+        - metavariable-regex:
+            metavariable: $MEMBER
+            regex: ((.*)-compute@developer.gserviceaccount.com)
```
Comment (minusworld) [inline terraform/gcp/security/gcp-org-member-default-service-account-iam-binding.yaml:13]: What part of this regex indicates that this is the "default" service account? I'm not familiar with GCP service accounts.

File: terraform/gcp/security/gcp-org-member-default-service-account-iam-member.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_organization_iam_member" "..." {
+              ...
+              member = $MEMBER
+              ...
+              }
+        - metavariable-regex:
+            metavariable: $MEMBER
+            regex: ((.*)-compute@developer.gserviceaccount.com)
```
Comment (minusworld) [inline terraform/gcp/security/gcp-org-member-default-service-account-iam-member.yaml:13]: Same question as above

File: terraform/gcp/security/gcp-project-member-default-service-account-iam-binding.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_project_iam_binding" "..." {
+              ...
+              members = [..., $MEMBER, ...]
+              ...
+              }
+        - metavariable-regex:
+            metavariable: $MEMBER
+            regex: ((.*)-compute@developer.gserviceaccount.com)
```
Comment (minusworld) [inline terraform/gcp/security/gcp-project-member-default-service-account-iam-binding.yaml:13]: Same question as above

File: terraform/gcp/security/gcp-project-member-default-service-account-iam-member.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        - pattern: resource
+        - pattern-inside: |
+              resource "google_project_iam_member" "..." {
+              ...
+              member = $MEMBER
+              ...
+              }
+        - metavariable-regex:
+            metavariable: $MEMBER
+            regex: ((.*)-compute@developer.gserviceaccount.com)
```
Comment (minusworld) [inline terraform/gcp/security/gcp-project-member-default-service-account-iam-member.yaml:13]: Same as above

File: terraform/gcp/security/gcp-sqlserver-no-public-ip.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+rules:
+- id: gcp-sqlserver-no-public-ip
+  patterns:
+    - pattern: resource
+    - pattern-inside: |
+        resource "google_sql_database_instance" "..." {
+        ...
+        ip_configuration {
+          ...
+          ipv4_enabled = true
```
Comment (minusworld) [inline terraform/gcp/security/gcp-sqlserver-no-public-ip.yaml:10]: Is this really all that's required to give the DB instance a public IP? 😮  Crazy

**Review by minusworld — CHANGES_REQUESTED** [review]
Left questions above

File: terraform/gcp/security/gcp-org-impersonation-roles-iam-binding.yaml (hunk 1 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-org-impersonation-roles-iam-binding.yaml:9]: Yes, the issue is with the editor role (https://cloud.google.com/iam/docs/understanding-roles).

File: terraform/gcp/security/gcp-org-impersonation-roles-iam-member.yaml (hunk 2 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-org-impersonation-roles-iam-member.yaml:9]: Yes, the issue is with the editor role (https://cloud.google.com/iam/docs/understanding-roles).

File: terraform/gcp/security/gcp-org-member-default-service-account-iam-binding.yaml (hunk 3 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-org-member-default-service-account-iam-binding.yaml:13]: The default service account structure is `PROJECT_NUMBER-compute@developer.gserviceaccount.com` (https://cloud.google.com/compute/docs/access/service-accounts#default_service_account).

File: terraform/gcp/security/gcp-org-member-default-service-account-iam-member.yaml (hunk 4 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-org-member-default-service-account-iam-member.yaml:13]: The default service account structure is `PROJECT_NUMBER-compute@developer.gserviceaccount.com` (https://cloud.google.com/compute/docs/access/service-accounts#default_service_account).

File: terraform/gcp/security/gcp-project-member-default-service-account-iam-binding.yaml (hunk 5 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-project-member-default-service-account-iam-binding.yaml:13]: The default service account structure is `PROJECT_NUMBER-compute@developer.gserviceaccount.com` (https://cloud.google.com/compute/docs/access/service-accounts#default_service_account).

File: terraform/gcp/security/gcp-project-member-default-service-account-iam-member.yaml (hunk 6 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-project-member-default-service-account-iam-member.yaml:13]: The default service account structure is `PROJECT_NUMBER-compute@developer.gserviceaccount.com` (https://cloud.google.com/compute/docs/access/service-accounts#default_service_account).

File: terraform/gcp/security/gcp-sqlserver-no-public-ip.yaml (hunk 7 above)
Comment (mertcancoskuner) [reply terraform/gcp/security/gcp-sqlserver-no-public-ip.yaml:10]: According to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#ipv4_enabled, yes.

**Comment by minusworld** [comment]
@mertcancoskuner Looks like tests are failing on `gcp-postgresql-log-min-message`, I assume because of the regular expression. Could you noodle on the rule to make tests pass? I started a snippet, here: https://semgrep.dev/s/k12e

**Comment by mertcancoskuner** [comment]
> @mertcancoskuner Looks like tests are failing on `gcp-postgresql-log-min-message`, I assume because of the regular expression. Could you noodle on the rule to make tests pass? I started a snippet, here: https://semgrep.dev/s/k12e

According to https://regex101.com/r/peCUYv/1 test cases should match now. However, https://semgrep.dev/playground/s/mertcancoskuner:gcp-postgresql-log-min-message-copy missing them when used with `pattern-inside`, `pattern-not-inside`.

**Comment by minusworld** [comment]
> According to https://regex101.com/r/peCUYv/1 test cases should match now. However, https://semgrep.dev/playground/s/mertcancoskuner:gcp-postgresql-log-min-message-copy missing them when used with `pattern-inside`, `pattern-not-inside`.

Ah, gotcha. I was able to switch things up a bit to produce this, https://semgrep.dev/s/BqwR, which looks like it matches correctly. I can update this tomorrow, or you can if you beat me to it tonight. :)

**Comment by mertcancoskuner** [comment]
> > According to https://regex101.com/r/peCUYv/1 test cases should match now. However, https://semgrep.dev/playground/s/mertcancoskuner:gcp-postgresql-log-min-message-copy missing them when used with `pattern-inside`, `pattern-not-inside`.
> 
> Ah, gotcha. I was able to switch things up a bit to produce this, https://semgrep.dev/s/BqwR, which looks like it matches correctly. I can update this tomorrow, or you can if you beat me to it tonight. :)

I updated the rule!

**Review by minusworld — APPROVED** [review]

**Pull request merged**